SofascoreAcademy/SofascoreAcademy/View/SportsTabView.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  SportsTabView.swift
+//  SofascoreAcademy
+//
+//  Created by Akademija on 07.04.2024..
+//
+
+import UIKit
+import SofaAcademic
+
+class SportsTabView: BaseView {
+    
+    private let horizontalStackView: UIStackView = .init()
+    private var tabItemViews: [SportTabItemView] = .init()
+    
+    private var currentSelectedTab: SportTabItemView?
+    private var changedTabHandler: ((Sport) -> Void)? = nil
+
+    override init() {
+        super.init()
+    }
+    
+    override func addViews() {
+        for sport in Sport.allCases {
+            let tabItemView: SportTabItemView = .init()
+            tabItemView.tag = sport.rawValue
+            tabItemViews.append(tabItemView)
+            horizontalStackView.addArrangedSubview(tabItemView)
+        }
+        
+        addSubview(horizontalStackView)
+    }
+    
+    override func styleViews() {
+        backgroundColor = .primaryDefault
+        horizontalStackView.distribution = .fillEqually
+        
+        for tabItemView in tabItemViews {
+            let sport: Sport = .getSportForValue(tabItemView.tag)
+            tabItemView
+                .sportName(sport.name)
+                .sportImage(sport.image)
+            
+            if UserDefaults.selectedTab == tabItemView.tag {
+                currentSelectedTab = tabItemView
+            }
+        }
+        
+        currentSelectedTab?.setSelected(true)
+    }
+    
+    override func setupConstraints() {
+        horizontalStackView.snp.makeConstraints() {
+            $0.edges.equalToSuperview()
+        }
+    }
+    
+    override func setupGestureRecognizers() {
+        for tabItemView in tabItemViews {
+            tabItemView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(tabSelected)))
+        }
+    }
+    
+    @objc
+    private func tabSelected(_ sender: UITapGestureRecognizer) {
+        currentSelectedTab?.setSelected(false)
+        currentSelectedTab = sender.view as? SportTabItemView
+        currentSelectedTab?.setSelected(true)
+        
+        let index: Int = currentSelectedTab?.tag ?? 0
+        
+        UserDefaults.setSelectedTabIndex(index)
+        changedTabHandler?(.getSportForValue(index))
+    }
```

Comment:
Ovo bi trbealo biti rješeno s update funkcijom od gore. 
View mora biti toliko glup da ne zna što treba napraviti kada se klikne na neki njegov element... znamo samo što mora vratiti... i onda u VC napraviš akciju i zoveš update nad ovim viewom.